src/main/java/team4/footwithme/member/jwt/JwtTokenUtil.java

```diff
@@ -0,0 +1,169 @@
+    }
+
+    public TokenResponse createToken(String email){
+        String accessToken = createAccessToken(email);
+        String refreshToken = createRefreshToken(email);
+
+        return TokenResponse.of(accessToken, refreshToken, REFRESH_TIME);
+    }
+
+    public String createAccessToken(String email) {
+        Date date = new Date();
+        MemberRole role = getRoleFromEmail(email);
+
+        return Jwts.builder()
+                .setSubject(email)
+                .claim("role", role.name())
+                .setExpiration(new Date(date.getTime() + ACCESS_TIME))
+                .setIssuedAt(date)
+                .signWith(SignatureAlgorithm.HS256, key)
+                .compact();
+    }
+
+    public String reCreateAccessToken(String refreshToken){
+        Date date = new Date();
+        String email = getEmailFromToken(refreshToken);
+        MemberRole role = getRoleFromEmail(email);
+
+        return Jwts.builder()
+                .setSubject(email)
+                .claim("role", role.name())
+                .setExpiration(new Date(date.getTime() + ACCESS_TIME))
+                .setIssuedAt(date)
+                .signWith(key, SignatureAlgorithm.HS256)
+                .compact();
+    }
+
+    public String createRefreshToken(String email){
+        Date date = new Date();
+
+        String refreshToken = Jwts.builder()
+                .setSubject(email)
+                .setExpiration(new Date(date.getTime() + REFRESH_TIME))
+                .setIssuedAt(date)
+                .signWith(key, SignatureAlgorithm.HS256)
+                .compact();
+
+        return refreshToken;
+    }
+
+    public Authentication createAuthentication(String email) {
+        UserDetails userDetails = userDetailService.loadUserByUsername(email);
+        return new UsernamePasswordAuthenticationToken(userDetails, "", userDetails.getAuthorities());
+    }
+
+    public Long getExpiration(String accessToken) {
+        Date expiration = Jwts.parserBuilder().setSigningKey(key).build().parseClaimsJws(accessToken).getBody().getExpiration();
+
+        Long now = new Date().getTime();
+        return (expiration.getTime() - now);
+    }
+
+    private MemberRole getRoleFromEmail(String email) {
+        Member member = memberRepository.findByEmail(email)
+                .orElseThrow(() -> new IllegalArgumentException("존재하지 않는 사용자 입니다."));
+
+        return member.getMemberRole();
+    }
+
+    public String getEmailFromToken(String token) {
+        return Jwts.parserBuilder().setSigningKey(key).build().parseClaimsJws(token).getBody().getSubject();
+    }
+
+    public String resolveToken(HttpServletRequest request) {
+        String token = request.getHeader(HttpHeaders.AUTHORIZATION);
+
+        if(StringUtils.hasText(token)) {
+            if(token.startsWith(BEARER_PREFIX)) {
+                return token.substring(7).trim();
+            } else {
+                return token;
+            }
+        }
```

Comment:
```suggestion
        if(StringUtils.hasText(token)) {
            if(token.startsWith(BEARER_PREFIX)) {
                return token.substring(7).trim();
            }
            return token;
            }
        }
```
아마도 이런 방식으로 얼리리턴을 사용하는게 더 좋을 수도 있을 것 같아요

else를 사용하기 보다 얼리리턴을 사용하는게 좋다고 합니다 !
한번 아래 블로그 참고해봐도 좋을 것 같아요

[얼리 리턴 패턴(Early Return Pattern)](https://velog.io/@orodae/%EC%96%BC%EB%A6%AC-%EB%A6%AC%EC%96%BC-%ED%8C%A8%ED%84%B4Early-Return-Pattern)

Reply:
아하 넵 감사합니다!